tests/unit/test_report_generator_unit.py

```
@@ -0,0 +1,666 @@
+    info = generator.get_template_info(None)
+    assert info["valid"] is True
+    assert info["name"] == template_config.name
+
+
+def test_show_progress_indicator_respects_stop_event() -> None:
+    generator = ReportGenerator()
+    stop_event = threading.Event()
+
+    thread = threading.Thread(target=generator._show_progress_indicator, args=(stop_event, 1))
+    thread.start()
+    import time
```

Ruff (0.14.1)

Redefinition of unused `time` from line 7

Remove definition: `time`

(F811)